src/borg/helpers/fs.py

```diff
@@ -230,6 +231,9 @@ def os_open(*, flags, path=None, parent_fd=None, name=None, noatime=False):
         fname = name  # use name relative to parent_fd
     else:
         fname, parent_fd = path, None  # just use the path
+    if is_win32 and os.path.isdir(fname):
+        # Directories can not be opened on Windows.
+        return None
```

Comment:
Note: Currently I don't understand the full consequences of this. As seen in the other changes, I made the code aware of file descriptors which are potentialy `None` and a `borg create` and extract seems to work.

Reply:
@ThomasWaldmann do you see any problems with this?

Reply:
No. It will either work (because you added code to deal with that) or very obviously blow up if some place still tries to do operations on a fd that is None.